chainer/functions/connection/convolution_nd.py

```diff
@@ -173,17 +175,64 @@ def forward(self, inputs):
         else:
             return self._forward_cudnn(x, W, b)
 
-    def _backward_xp(self, x, W, b, gy, xp):
-        dims = x.shape[2:]     # (n, c_I, d_1, d_2, ..., d_N)
-        stride = self.stride
-        pad = self.pad
-        ndim = self.ndim
+    def backward(self, indexes, grad_outputs):
+        x, W = self.get_retained_inputs()
+        gy, = grad_outputs
+
+        ret = []
+        if 0 in indexes:
+            x_shape = x.shape[2:]
+            gx = chainer.functions.deconvolution_nd(
+                gy, W, stride=self.stride, pad=self.pad, outsize=x_shape)
+            ret.append(gx)
+        if 1 in indexes:
+            gW, = ConvolutionNDGradW(self).apply((x, gy))
+            ret.append(gW)
+        if 2 in indexes:
+            axis = (0,) + tuple(moves.range(2, gy.ndim))
+            gb = chainer.functions.sum(gy, axis=axis)
+            ret.append(gb)
+
+        return ret
+
+
+class ConvolutionNDGradW(function_node.FunctionNode):
+
+    def __init__(self, convnd):
```

Comment:
How about passing each argument separately? I think its interface is better aligned with `ConvolutionND.__init__` that way.

Reply:
This is because I simply followed the design of API in `Convolution2D`. I think either is fine because it is an internal design problem that does not need to be released (it is necessary to adopt to either, though).

Reply:
Ah, I see, then I agree we should leave it this way. It's not really labelled as a non-public API so I personally think we should think it through but it should of course be consistent.